src/Symfony/Component/Validator/Constraints/NotBlankValidator.php

```diff
@@ -29,7 +29,9 @@ public function validate($value, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\NotBlank');
         }
 
-        if (false === $value || (empty($value) && '0' != $value)) {
+        $normalizedValue = $constraint->trim ? trim($value) : $value;
```

Comment:
You need to check whether the value is a string in this validator - someone may use the option incorrectly and `trim()` expects it.

Also, would be nice if you added tests for Unicode whitespaces, e.g. `\u{00A0} test` - I'm pretty sure that in PHP 7.1.x both `trim()` etc. and `\s` regex meta character don't consider them still. See: https://stackoverflow.com/questions/4166896/trim-unicode-whitespace-in-php-5-2

Reply:
Thanks @Destroy666x. Now the regex also works for every Unicode whitespace.